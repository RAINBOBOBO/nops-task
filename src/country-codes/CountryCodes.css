.buttonA {
  background-color: #46139f
}

.buttonB {
  background-color: #ff7f50
}