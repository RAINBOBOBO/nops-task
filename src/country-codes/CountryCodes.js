import React from "react";
import { useHistory } from "react-router-dom";
import "./CountryCodes.css"

function CountryCodes() {
  const history = useHistory();

  function openModalA() {
    history.push("/codes/a");
  }

  function openModalB() {
    history.push("/codes/b");
  }

  return (
    <div>
      <button
        name="buttonA"
        className="buttonA"
        onClick={openModalA}
        data-testid="button-a"
      >
        Button A
      </button>

      <button
        name="buttonB"
        className="buttonB"
        onClick={openModalB}
        data-testid="button-b"
      >
        Button B
      </button>
    </div>
  );
}

export default CountryCodes;