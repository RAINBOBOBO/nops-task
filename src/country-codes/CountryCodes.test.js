import React from "react";
import { render, fireEvent } from "@testing-library/react";
import CountryCodes from "./CountryCodes";
import { MemoryRouter } from "react-router";

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <CountryCodes />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

