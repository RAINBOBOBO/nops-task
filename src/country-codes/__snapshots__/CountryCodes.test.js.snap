// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <button
      class="btn btn-primary"
      data-testid="button-a"
      name="buttonA"
    >
      Button A
    </button>
    <button
      class="btn btn-primary"
      data-testid="button-b"
      name="buttonB"
    >
      Button B
    </button>
  </div>
</DocumentFragment>
`;
